bcgov/bcparks.ca
# CM-412: Add robots txt and sitemap

### Jira Ticket:
CM-412

### Jira Ticket URL:
https://bcparksdigital.atlassian.net/browse/CM-412

### Description:
- Added `gatsby-plugin-sitemap` to generate `sitemap.xml` file
- Added `gatsby-plugin-robots-txt` to generate `robots.txt` file
- Fixed canonical URL bug
- Ref: https://www.gatsbyjs.com/plugins/gatsby-plugin-robots-txt / https://www.gatsbyjs.com/plugins/gatsby-plugin-sitemap

### Testing:
1. `yarn`
2. Add `REACT_APP_SITE_URL=http://localhost:8000` to your `.env`
3. `gatsby clean`
4. `gatby build`
5. Check if there's a `robots.txt` at `/publc`
6. Check if there's a `/sitemap` at `/publc`
<img width="265" alt="Screen Shot 2023-02-02 at 4 23 22 PM" src="https://user-images.githubusercontent.com/90415568/216481156-bf4a82fe-57a4-493d-b961-357a42470f32.png">
7. Check if the canonical URL uses `REACT_APP_SITE_URL` on Google console

```
<link rel="canonical" key="http://localhost:8000/" href="http://localhost:8000/" data-react-helmet="true">
```
@molund 
Seems the default `robots.txt` generated by `gatsby-plugin-robots-txt` has always `Sitemap` and `Host`...
Not sure if we can have only the first 2 lines
```
User-agent: *
Allow: /
Sitemap: http://localhost:1337/sitemap-index.xml
Host: localhost:1337
```
* This part `localhost:1337` will be changed dynamically based on env
